src/utils/transforms.cljs

```diff
@@ -21,3 +24,37 @@
   [json]
   (when-not (= json "undefined")
     (try (.parse js/JSON json) (catch js/Error _ (when (string? json) json)))))
+
+;; Functions took from reagent and modified to also transform vectors contained in styles
+
+(def ^:private js-val? reagent.impl.util/js-val?)
+(def ^:private named? reagent.impl.util/named?)
+(def ^:private cached-prop-name reagent.impl.template/cached-prop-name)
+(declare styles-with-vectors)
+
+(defn ^:private kv-conv
+  [o k v]
+  (doto o
+    (oops/gobj-set (cached-prop-name k) (styles-with-vectors v))))
+
+(defn styles-with-vectors
+  [x]
+  (cond (js-val? x) x
+        (named? x)  (name x)
+        (map? x)    (reduce-kv kv-conv #js {} x)
+        (vector? x) (to-array (map styles-with-vectors x))
+        (coll? x)   (clj->js x)
+        (ifn? x)    (fn [& args]
+                      (apply x args))
+        :else       (clj->js x)))
+
+(defn copy-js-obj-to-map
+  "Copy `obj` keys and values into `m` if `(pred obj-key)` is satisfied."
+  [obj m pred]
+  (persistent!
+   (reduce (fn [acc js-prop]
+             (if (pred js-prop)
```

Comment:
Oh! I should have mention this before!

I already tried it, actually it was my first approach, but when tried, for some reason the keys couldn't be erased.
I used `cljs.core/js-delete`, it returned `false` and the object wasn't mutated.
ref: https://cljs.github.io/api/cljs.core/js-delete

Reply:
Will try it with `remove` from `goog.object`

Reply:
Interesting @ulisesmac, thanks for the context. `goog.object/remove` should work because the implementation uses `delete` from Javascript, which nukes any key in the JS object.

Reply:
It didn't work too :disappointed: 
BTW, `js-delete` at the end is just a macro using the `delete` from JS too.

Did an example to show it:
```clojure
;; Using a custom object:
(let [component #js {:props #js{:argv  "1"
                                :other "2"}}
      props     (oops/gobj-get component "props")]

  [(js-keys props)
   (gobj/remove props "argv")
   (js-keys props)
   (js-delete props "argv")
   (js-keys props)])
;; => 
[#js["argv" "other"] true #js["other"] true #js["other"]]
   
 ;; Using a REPL captured value in reanimated/view:
 (let [component --cu
       props     (oops/gobj-get component "props")]

  [(js-keys props)
   (gobj/remove props "argv")
   (js-keys props)
   (js-delete props "argv")
   (js-keys props)])
;;=> 
[#js["argv"] true #js["argv"] false #js["argv"]]
```

Weird that `goog.object` reports `true` but does nothinig.

Reply:
@ulisesmac I think in this case in the original code `external-props` is not what we think it is. If you try to inspect this value you get this error related to reanimated `Reading _value directly is only possible on the UI runtime.`, for example with `(gobj/unsafeClone external-props)`.

That's likely related to why `delete` doesn't do what's supposed to do. I've had problems in the past where `delete` would behave like a no-op, I think this is documented.

Reply:
Yes, the problem is that Reanimated value, it happens all the time with shared-values.

If you remove where it's stored, then we can just print it in a REPL, and yeah, maybe that's the reason.